### ISSUE TITLE
dependency injection exception instead of null

### ISSUE THREAD

**Issue by OneCyrus** [Azure/azure-functions-dotnet-extensions#34]
we try to add azure functions support to a graphql library which usually runs in asp.net core. this library strongly builds on top of the microsoft dependency injection. a lot of things work very smoothly but there are some issues with the way this extension throws exceptions instead of returning null.

i created a reproduction which demonstrates this issue. the code is at https://github.com/OneCyrus/AzF-DependencyInjection-Repro and it features a simple asp.net core webapi which returns a string "still works" (http://localhost:5000/api/inject)

there's also an azure function which does the same but throws an exception and only returns a 500 server error (http://localhost:7071/api/Inject).

I've also created an additional demo which shows the different behaviour of the DI in asp.net core with GetService and GetRequiredService. (http://localhost:5000/api/injectAdv)

**Comment by OneCyrus** [comment]
the specification of IServiceProvider defines it as follow:

> Returns
> Object
> A service object of type serviceType.
> 
> -or-
> 
> **null if there is no service object of type serviceType.**
> 

https://docs.microsoft.com/en-us/dotnet/api/system.iserviceprovider.getservice?view=netcore-3.1

**Comment by OneCyrus** [comment]
another strange thing is the dependency on `Microsoft.Extensions.DependencyInjection` here: https://github.com/Azure/azure-functions-dotnet-extensions/blob/ce4e8c43ed6372e0b14958128df79baa08abcf6d/src/Extensions/Extensions.csproj#L28

this is actually not used at all. the DI is based on a custom DI implementation in the azure functions host.

**Comment by galvesribeiro** [comment]
I completely agree this is totally wrong. If you inject the `IServiceProvider` and call `GetService()` it return null:

![image](https://user-images.githubusercontent.com/4714040/79071511-8116cf00-7cb2-11ea-816a-0bf63f8400dc.png)

With that in mind, I believe it is the host that is throwing and not the actual service provider...

**Comment by galvesribeiro** [comment]
@fabiocav do you this issue should be opened on the Host repo or here?

**Comment by fabiocav** [comment]
Apologies for the late response here.

This is indeed a host behavior and we just closed a similar issue with information about the plans [here](https://github.com/Azure/azure-functions-host/issues/6587#issuecomment-1949527312).

Closing this as "won't fix", as we won't be changing the behavior of the in-proc experience, but the comment linked above provides more details about the plans moving forward.
